### PR TITLE
[14.0][MIG] rma_repair

### PR DESCRIPTION
**Pull request by MateuGForgeFlow**
Standard migration to v14.
Depends on #147 #146 and https://github.com/OCA/manufacture/pull/577

cc @AaronHForgeFlow

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@MateuGForgeFlow can you rebase?

**Review by AaronHForgeFlow — APPROVED** [review]

**Comment by max3903** [comment]
:+1:

**Review by max3903 — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
@MateuGForgeFlow is this ready for merge or you are pending to include anything else?

File: rma_repair/models/rma_order_line.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            if rec.delivery_policy == "repair":
+                qty_to_pay = 0.0
+                for repair in rec.repair_ids.filtered(
+                    lambda r: r.invoice_method != "none"
+                    and r.invoice_status != "posted"
+                ):
+                    qty_to_pay += repair.product_qty
+                rec.qty_to_pay = qty_to_pay
+            else:
+                rec.qty_to_pay = 0.0
```
Comment (bodedra) [inline rma_repair/models/rma_order_line.py]:
```suggestion
            qty_to_pay = 0.0
            if rec.delivery_policy == "repair":
                for repair in rec.repair_ids.filtered(
                    lambda r: r.invoice_method != "none"
                    and r.invoice_status != "posted"
                ):
                    qty_to_pay += repair.product_qty
            rec.qty_to_pay = qty_to_pay
```

File: rma_repair/wizards/rma_order_line_make_repair.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        assert active_model == "rma.order.line", "Bad context propagation"
+        items = []
+        lines = rma_line_obj.browse(rma_line_ids)
+        for line in lines:
+            items.append([0, 0, self._prepare_item(line)])
+        res["item_ids"] = items
+        return res
+
+    def make_repair_order(self):
+        res = []
```
Comment (bodedra) [inline rma_repair/wizards/rma_order_line_make_repair.py:59]:
```suggestion
        self.ensure_one()
        res = []
```

File: rma_repair/wizards/rma_order_line_make_repair.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        required=True,
+        help="Selecting 'Before Repair' or 'After Repair' will allow you "
+        "to generate invoice before or after the repair is done "
+        "respectively. 'No invoice' means you don't want to generate "
+        "invoice for this repair order.",
+    )
+
+    @api.model
+    def _prepare_repair_order(self, rma_line):
+        location_dest = (
```
Comment (bodedra) [inline rma_repair/wizards/rma_order_line_make_repair.py:148]:
```suggestion
        self.ensure_one()
        location_dest = (
```

**Review by bodedra — APPROVED** [review]
Added minor suggestion

File: rma_repair/wizards/rma_order_line_make_repair.py [hunk 4]
```diff
@@ -144,6 +145,7 @@ def _onchange_to_refurbish(self):
 
     @api.model
```
Comment (AaronHForgeFlow) [inline rma_repair/wizards/rma_order_line_make_repair.py]: here remove the api.model, because we are actually passing a record

**Pull request merged**